docs/book/src/developer/providers/contracts/infra-machinepool.md

```diff
@@ -0,0 +1,525 @@
+
+If you implement this then create a `status.instances` field that is a slice of a struct type that contains the information you want to store and be made available to the users.
+
+```go
+type FooMachinePoolStatus struct {
+    // Instances contains the status for each instance in the pool
+    // +optional
+    Instances []FooMachinePoolInstanceStatus `json:"instances,omitempty"`
+    
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool status.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+
+// FooMachinePoolInstanceStatus contains instance status information about a FooMachinePool.
+type FooMachinePoolInstanceStatus struct {
+ // Addresses contains the associated addresses for the machine.
+ // +optional
+ Addresses []clusterv1.MachineAddress `json:"addresses,omitempty"`
+
+ // InstanceName is the identification of the Machine Instance within the Machine Pool
+ InstanceName string `json:"instanceName,omitempty"`
+
+ // ProviderID is the provider identification of the Machine Pool Instance
+ // +optional
+ ProviderID *string `json:"providerID,omitempty"`
+
+ // Version defines the Kubernetes version for the Machine Instance
+ // +optional
+ Version *string `json:"version,omitempty"`
+
+ // Ready denotes that the machine is ready
+ // +optional
+ Ready bool `json:"ready"`
+}
+```
+
+### InfraMachinePool: providerID
+
+Each InfraMachinePool MAY specify a provider ID on `spec.providerID` that can be used to identify the InfraMachinePool.
+
+```go
+type FooMachinePoolSpec struct {
+    // providerID is the identification ID of the FooMachinePool.
+    // +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=512
+	ProviderID string `json:"providerID,omitempty"`
+    
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool spec.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+```
+
+NOTE: To align with API conventions, we recommend since the v1beta2 contract that the `ProviderID` field should be
+of type `string`.
+
+
+### InfraMachinePool: providerIDList
+
+Each InfraMachinePool MUST supply a list of the identification IDs of the machine instances managed by the machine pool by storing these in `spec.providerIDList`.
+
+```go
+type FooMachinePoolSpec struct {
+    // ProviderIDList is the list of identification IDs of machine instances managed by this Machine Pool
+	// +optional
+	ProviderIDList []string `json:"providerIDList,omitempty"`
+    
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool spec.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+```
+
+Cluster API uses this list to determine the status of the machine pool and to know when replicas have been deleted, at which point the Node will be deleted.
+
+### InfraMachinePool: ready
```

Comment:
I would suggest to call this `InfraMachinePool: initialization completed` and chart a part to move from `status.ready` to `status.initialization.provisioned`, thus aligning to other CAPI resources (see https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20240916-improve-status-in-CAPI-resources.md#machinepool-newconditions).

Notably, f the change is implemented ASAP, for the transition you can leverage the v1beta1 compatibility support already in place for other CAPI resources